File: backend/internal/service/dto.go
package service

import (
	"github.com/Sakrafux/family-tree-app/backend/internal/db"
	"github.com/google/uuid"
)

type SpouseDto struct {
	Id         uuid.UUID
	SinceYear  *int32
	SinceMonth *int32
	SinceDay   *int32
	UntilYear  *int32
	UntilMonth *int32
	UntilDay   *int32
}

type SiblingDto struct {
	Id     uuid.UUID
	IsHalf bool
}

type PersonDto struct {
	*db.Person
	Age      *int32
	Level    int
	Distance int64
	Parents  []uuid.UUID
	Children []uuid.UUID
	Siblings []SiblingDto
	Spouses  []SpouseDto
}

type FamilyTreeDto struct {
	Root    *PersonDto
	Persons map[uuid.UUID]*PersonDto
}

type PostFeedbackRequest struct {
	Text string
}

type PatchFeedbackResolveRequest struct {
	IsResolved bool
}

type FeedbackDto struct {
	*db.Feedback
}

type LoginRequest struct {
	Username string
	Password string
}

type AccessTokenDto struct {
	AccessToken string
}

File: frontend/src/types/dto.ts
export type SpouseDto = {
    Id: string;
    SinceYear?: number;
    SinceMonth?: number;
    SinceDay?: number;
    UntilYear?: number;
    UntilMonth?: number;
    UntilDay?: number;
};

export type SiblingDto = {
    Id: string;
    IsHalf: boolean;
};

export type PersonDto = {
    Id: string;
    FirstName?: string;
    MiddleName?: string;
    LastName?: string;
    BirthName?: string;
    Gender?: string;
    IsDead?: boolean;
    BirthDateYear?: number;
    BirthDateMonth?: number;
    BirthDateDay?: number;
    DeathDateYear?: number;
    DeathDateMonth?: number;
    DeathDateDay?: number;
    Level: number;
    Distance: number;
    Parents: string[];
    Children: string[];
    Siblings: SiblingDto[];
    Spouses: SpouseDto[];
};

export type FamilyTreeDto = {
    Root: PersonDto;
    Persons: Record<string, PersonDto>;
};

export type FeedbackDto = {
    Id: number;
    Text: string;
    Timestamp: string;
};
